quarkchain/cluster/tests/test_failed_transactions_gas_fee.py

```diff
@@ -0,0 +1,112 @@
+import unittest
+
+from quarkchain.cluster.shard_state import ShardState
+from quarkchain.cluster.tests.test_utils import (
+    get_test_env,
+    create_transfer_transaction,
+    contract_creation_tx,
+)
+
+from quarkchain.core import Identity, Address
+from quarkchain.core import CrossShardTransactionDeposit, CrossShardTransactionList
+from quarkchain.evm import opcodes
+from quarkchain.evm.messages import mk_contract_address
+from quarkchain.utils import token_id_encode
+from quarkchain.genesis import GenesisManager
+from quarkchain.cluster.tests.test_native_token import create_default_shard_state
+
+
+class TestFailTransactionGasCalucation(unittest.TestCase):
```

Comment:
recommend to add this test to either `quarkchain/evm/tests/test_state.py` or `quarkchain/cluster/tests/test_shard_state.py`

Reply:
I will do it.